#EXPRESSION
Use Cufflinks to generate expression estimates from the SAM/BAM files generated by TopHat in the previous module
	
###Note on de novo transcript discovery and differential expression using Cufflinks and Cuffdiff
In this module, we will run Cufflinks in 'reference only' mode. For simplicity and to reduce run time, it is sometime useful to perform expression analysis with only known transcript models. However, Cufflinks can predict the transcripts present in each library instead (by dropping the '-G' option in cufflinks commands as described later in Module 4). Cufflinks will then assign arbitrary transcript IDs to each transcript assembled from the data and estimate expression for those transcripts. One complication with this method is that in each library a different set of transcripts is likely to be predicted for each library. There may be a lot of similarities but the number of transcripts and their exact structure will differ in the output files for each library. Before you can compare across libraries you therefore need to determine which transcripts correspond to each other across the libraries.

* Cufflinks provides 'cuffmerge' to combine predicted transcript GTF files from across different libraries
 * Once you have a merged GTF file you can run Cuffdiff with this instead of the known transcripts GTF file we used above
* Cufflinks also provides 'cuffcompare' to compare predicted transcripts to known transcripts
* Refer to the Cufflinks manual for a more detailed explanation:
 * http://cufflinks.cbcb.umd.edu/manual.html
	
Cufflinks basic usage:  
```
cufflinks [options] <hits.sam>
```
	
Extra options specified below:  
'-p 8' tells Cufflinks to use eight CPUs  
'-G/--GTF <known transcripts file>' Forces cufflinks to calculate expression values for just known transcripts (we call this "reference only" mode)  
'-o' tells Cufflinks to write output to a particular directory (one per sample)  

	cd $RNA_HOME/
	mkdir -p expression/tophat_cufflinks/ref_only/
	cd expression/tophat_cufflinks/ref_only/
	cufflinks -p 8 -o HBR_Rep1_ERCC-Mix2 --GTF $RNA_HOME/refs/hg19/genes/genes_chr22_ERCC92.gtf --frag-len-mean 262 --frag-len-std-dev 80 --no-update-check $RNA_HOME/alignments/tophat/HBR_Rep1_ERCC-Mix2/accepted_hits.bam
	cufflinks -p 8 -o HBR_Rep2_ERCC-Mix2 --GTF $RNA_HOME/refs/hg19/genes/genes_chr22_ERCC92.gtf --frag-len-mean 262 --frag-len-std-dev 80 --no-update-check $RNA_HOME/alignments/tophat/HBR_Rep2_ERCC-Mix2/accepted_hits.bam
	cufflinks -p 8 -o HBR_Rep3_ERCC-Mix2 --GTF $RNA_HOME/refs/hg19/genes/genes_chr22_ERCC92.gtf --frag-len-mean 262 --frag-len-std-dev 80 --no-update-check $RNA_HOME/alignments/tophat/HBR_Rep3_ERCC-Mix2/accepted_hits.bam

	cufflinks -p 8 -o UHR_Rep1_ERCC-Mix1 --GTF $RNA_HOME/refs/hg19/genes/genes_chr22_ERCC92.gtf --frag-len-mean 262 --frag-len-std-dev 80 --no-update-check $RNA_HOME/alignments/tophat/UHR_Rep1_ERCC-Mix1/accepted_hits.bam
	cufflinks -p 8 -o UHR_Rep2_ERCC-Mix1 --GTF $RNA_HOME/refs/hg19/genes/genes_chr22_ERCC92.gtf --frag-len-mean 262 --frag-len-std-dev 80 --no-update-check $RNA_HOME/alignments/tophat/UHR_Rep2_ERCC-Mix1/accepted_hits.bam
	cufflinks -p 8 -o UHR_Rep3_ERCC-Mix1 --GTF $RNA_HOME/refs/hg19/genes/genes_chr22_ERCC92.gtf --frag-len-mean 262 --frag-len-std-dev 80 --no-update-check $RNA_HOME/alignments/tophat/UHR_Rep3_ERCC-Mix1/accepted_hits.bam
	
##OPTIONAL ALTERNATIVE
Run cufflinks on STAR alignments instead of TopHat alignments

  cd $RNA_HOME/
	mkdir -p expression/star_cufflinks/ref_only/
	cd expression/star_cufflinks/ref_only/
	cufflinks -p 8 -o Normal_cDNA1_lib2 --GTF $RNA_HOME/refs/hg19/genes/genes_chr22.gtf --frag-len-mean 262 --frag-len-std-dev 80 --no-update-check $RNA_HOME/alignments/star/Normal_cDNA1_lib2/Aligned.out.sorted.bam
	cufflinks -p 8 -o Normal_cDNA2_lib2 --GTF $RNA_HOME/refs/hg19/genes/genes_chr22.gtf --frag-len-mean 262 --frag-len-std-dev 80 --no-update-check $RNA_HOME/alignments/star/Normal_cDNA2_lib2/Aligned.out.sorted.bam
	cufflinks -p 8 -o Tumor_cDNA1_lib2 --GTF $RNA_HOME/refs/hg19/genes/genes_chr22.gtf --frag-len-mean 262 --frag-len-std-dev 80 --no-update-check $RNA_HOME/alignments/star/Tumor_cDNA1_lib2/Aligned.out.sorted.bam
	cufflinks -p 8 -o Tumor_cDNA2_lib2 --GTF $RNA_HOME/refs/hg19/genes/genes_chr22.gtf --frag-len-mean 262 --frag-len-std-dev 80 --no-update-check $RNA_HOME/alignments/star/Tumor_cDNA2_lib2/Aligned.out.sorted.bam
	
What does the raw output from Cufflinks look like?
	cd $RNA_HOME/expression/tophat_cufflinks/ref_only/Tumor_cDNA1_lib2/
	ls -l 
	head isoforms.fpkm_tracking
	cut -f 1,4,7-13 isoforms.fpkm_tracking | less

Press 'q' to exit the 'less' display
	
##OPTIONAL ALTERNATIVE
Run htseq-count on alignments instead to produce raw counts instead of FPKM values for differential expression analysis
	
Refer to the HTSeq documentation for a more detailed explanation:
http://www-huber.embl.de/users/anders/HTSeq/doc/count.html
	
 htseq-count basic usage:
 htseq-count [options] <sam_file> <gff_file>
	
Extra options specified below
 '--mode' determines how to deal with reads that overlap more than one feature. We believe the 'intersection-strict' mode is best.
 '--stranded' specifies whether data is stranded or not. 
 '--minaqual' will skip all reads with alignment quality lower than the given minimum value
 '--type' specifies the feature type (3rd column in GFF file) to be used. (default, suitable for RNA-Seq and Ensembl GTF files: exon)
 '--idattr' The feature ID used to identity the counts in the output table. The default, suitable for RNA-SEq and Ensembl GTF files, is gene_id.
 NOTE: Instead of supplying the SAM file to htseq-count, we specify "-" to tell it to accept a stream from stdout
	
First, we need name-sorted bam files (we could use either the tophat or STAR alignments)
We have chosen to use tophat alignments here:
	cd $RNA_HOME/alignments/tophat
	samtools sort -n Normal_cDNA1_lib2/accepted_hits.bam Normal_cDNA1_lib2/accepted_hits_namesorted
	samtools sort -n Normal_cDNA2_lib2/accepted_hits.bam Normal_cDNA2_lib2/accepted_hits_namesorted
	samtools sort -n Tumor_cDNA1_lib2/accepted_hits.bam Tumor_cDNA1_lib2/accepted_hits_namesorted
	samtools sort -n Tumor_cDNA2_lib2/accepted_hits.bam Tumor_cDNA2_lib2/accepted_hits_namesorted
	
Next use samtools to pipe sam-format from these bam files to htseq-count
	cd $RNA_HOME/
	mkdir -p expression/tophat_counts
	cd expression/tophat_counts
	mkdir Normal_cDNA1_lib2 Normal_cDNA2_lib2 Tumor_cDNA1_lib2 Tumor_cDNA2_lib2
	
Gene-level counts
	samtools view -h $RNA_HOME/alignments/tophat/Normal_cDNA1_lib2/accepted_hits_namesorted.bam | $RNA_HOME/tools/HTSeq-0.6.1p1/scripts/htseq-count --mode intersection-strict --stranded no --minaqual 1 --type exon --idattr gene_id - $RNA_HOME/refs/hg19/genes/genes_chr22.gtf > Normal_cDNA1_lib2/gene_read_counts_table.tsv 
	samtools view -h $RNA_HOME/alignments/tophat/Normal_cDNA2_lib2/accepted_hits_namesorted.bam | $RNA_HOME/tools/HTSeq-0.6.1p1/scripts/htseq-count --mode intersection-strict --stranded no --minaqual 1 --type exon --idattr gene_id - $RNA_HOME/refs/hg19/genes/genes_chr22.gtf > Normal_cDNA2_lib2/gene_read_counts_table.tsv 
	samtools view -h $RNA_HOME/alignments/tophat/Tumor_cDNA1_lib2/accepted_hits_namesorted.bam | $RNA_HOME/tools/HTSeq-0.6.1p1/scripts/htseq-count --mode intersection-strict --stranded no --minaqual 1 --type exon --idattr gene_id - $RNA_HOME/refs/hg19/genes/genes_chr22.gtf > Tumor_cDNA1_lib2/gene_read_counts_table.tsv 
	samtools view -h $RNA_HOME/alignments/tophat/Tumor_cDNA2_lib2/accepted_hits_namesorted.bam | $RNA_HOME/tools/HTSeq-0.6.1p1/scripts/htseq-count --mode intersection-strict --stranded no --minaqual 1 --type exon --idattr gene_id - $RNA_HOME/refs/hg19/genes/genes_chr22.gtf > Tumor_cDNA2_lib2/gene_read_counts_table.tsv 
	
Merge results files into a single matrix for use in edgeR
	join <(join Normal_cDNA1_lib2/gene_read_counts_table.tsv Normal_cDNA2_lib2/gene_read_counts_table.tsv) <(join Tumor_cDNA1_lib2/gene_read_counts_table.tsv Tumor_cDNA2_lib2/gene_read_counts_table.tsv) > gene_read_counts_table_all.tsv
