#### 1. Recommended Reading
- [Integrative Genomics Viewer (IGV): high-performance genomics data visualization and exploration] (https://github.com/griffithlab/rnaseq_tutorial/wiki/Reading/IGV_2012.pdf)
- [Differential gene and transcript expression analysis of RNA-seq experiments with TopHat and Cufflinks] (https://github.com/griffithlab/rnaseq_tutorial/wiki/Reading/ExpressionAnalysis_Cufflinks_Trapnell_2012.pdf)
- [Differential analysis of gene regulation at transcript resolution with RNA-seq] (https://github.com/griffithlab/rnaseq_tutorial/wiki/Reading/DifferentialExpressionAnalysis_CuffDiff2_Trapnell_2013.pdf)
- [Methods to study splicing from high-throughput RNA sequencing data] (https://github.com/griffithlab/rnaseq_tutorial/wiki/Reading/SplicingAnalysisMethods_2014.pdf)

#### 2. Unix/Linux Tutorials
- [A UNIX command cheat sheet](http://www.rain.org/~mkummel/unix.html)
- [A UNIX tutorial for beginners](http://www.ee.surrey.ac.uk/Teaching/Unix/)
- [A UNIX and Perl primer for biologists](http://korflab.ucdavis.edu/Unix_and_Perl/current.html)
- [A detailed Biostars thread describing resources for learning Unix](https://www.biostars.org/p/16315/)

#### 3. R Tutorials
- [An R command cheat sheet](https://github.com/griffithlab/rnaseq_tutorial/wiki/Reading/R-short-refcard.pdf)
- [The official R intro manual](http://cran.r-project.org/doc/manuals/r-release/R-intro.html)
- [Rseek, a search engine for R related questions](http://rseek.org/)
- [A detailed BioStars thread describing resources for learning R](https://www.biostars.org/p/539/)

#### 4. Cloud Tutorials
- [Setting up Amazon EC2](http://docs.aws.amazon.com/AWSEC2/latest/UserGuide/get-set-up-for-amazon-ec2.html)
- [Getting started with Amazon EC2 Linux instances](http://docs.aws.amazon.com/AWSEC2/latest/UserGuide/EC2_GetStarted.html)

#### 5. Supplementary Materials Relating to the Hands On Tutorials of this Course
- [Description of the ERCC spike-in controls used in the tutorial data] (https://github.com/griffithlab/rnaseq_tutorial/wiki/ResourceFiles/ERCC.pdf)
- [Description of the Human Brain Reference (HBR) sample used to generate the tutorial data] (https://github.com/griffithlab/rnaseq_tutorial/wiki/ResourceFiles/HBR.pdf)
- [Description of the Universal Human Reference (UHR) sample used to generate the tutorial data] (https://github.com/griffithlab/rnaseq_tutorial/wiki/ResourceFiles/UHR.pdf)

#### 6. Recommended Tools for RNA-seq analysis
- [A recent trimming tool 'skewer' with comparison to many other read trimmers](http://www.biomedcentral.com/1471-2105/15/182/abstract)
- [HTS Mappers - A list of read mappers](http://wwwdev.ebi.ac.uk/fg/hts_mappers/)

#### 7. Useful RNA-seq and Bioinformatics resources
- [The RNA-seq blog - recent developments in RNA-seq technology and analysis](http://www.rna-seqblog.com/)
- [ENCODE RNA-seq Standards](https://github.com/griffithlab/rnaseq_tutorial/wiki/ResourceFiles/ENCODE_RNA-seq_standards_v1.0.pdf)
- [The Bioinformatics Links Directory](http://bioinformatics.ca/links_directory/) 
- [The periodic table of bioinformatics - A list of popular bioinformatics resources and tools](http://elements.eaglegenomics.com/)

#### 8. Getting More Help
- [Biostar - A forum for the bioinformatics community](http://www.biostars.org/)
- [SEQanswers - A forum for the next generation sequencing community](http://seqanswers.com/)

