![RNA-seq Flowchart - Module 5](Images/RNA-seq_Flowchart5.png)

#VISUALIZING RESULTS AT THE COMMAND LINE
	
View the junctions.bed files created by TopHat
	cd $RNA_HOME/alignments/tophat/Tumor_cDNA1_lib2/
	head junctions.bed
	
View the merged GTF file from the 'de_novo' mode.  Remember this merged GTF file combines both normal and tumor (GTFs for each were also produced)
	cd $RNA_HOME/expression/tophat_cufflinks/de_novo/merged/
	head merged.gtf
	
For details on the format of these files, refer to the following links:
http://cufflinks.cbcb.umd.edu/manual.html#cuffmerge
http://cufflinks.cbcb.umd.edu/manual.html#class_codes
	
View the differential splicing, differential promoter usage and differential CDS results files:
For each results file, we sort the results by the p-value and view the top 10 isoforms, ignoring entries classified as 'LOWDATA'
	cd $RNA_HOME/de/tophat_cufflinks/de_novo/
	
	sort -k 12n splicing.diff | grep -v LOWDATA | head
	
	sort -k 12n promoters.diff | grep -v LOWDATA | head
	
	sort -k 12n cds.diff | grep -v LOWDATA | head
	
How many genes have at least one transcript assembled by Cufflinks in the 'de_novo' results?
	cd $RNA_HOME/expression/tophat_cufflinks/de_novo/merged/
	cat merged.gtf | perl -ne 'if ($_ =~ /gene_name\s\"(\w+)\"/){print "$1\n"}' | sort | uniq | wc -l
	
How many genes have at least one *novel* transcript assembled?
	grep "j" merged.gtf | perl -ne 'if ($_ =~ /gene_name\s\"(\w+)\"/){print "$1\n"}' | sort | uniq | wc -l
	
	
##VISUALIZING RESULTS IN THE IGV BROWSER
	
 junctions.bed files:
 View the tophat junctions.bed file generated all the way back in Module 2
 http://cbw##.ssh01.com/rnaseq/alignments/tophat/Tumor_cDNA1_lib2/junctions.bed
 Go to the gene 'CYTH4'
 Do you see the evidence for a novel exon?
	
 merged.gtf files
 View the grand merged.gtf files that were generated by each of the three Cufflinks modes: 'ref_only', 'ref_guided', 'de_novo'
 http://cbw##.ssh01.com/rnaseq/expression/tophat_cufflinks/ref_only/merged/merged.gtf
 http://cbw##.ssh01.com/rnaseq/expression/tophat_cufflinks/ref_guided/merged/merged.gtf
 http://cbw##.ssh01.com/rnaseq/expression/tophat_cufflinks/de_novo/merged/merged.gtf
 Go to this region: '22:44823240-44916000'
 Do you see the evidence for a novel single exon transcript that is found in 'de_novo' and 'ref_guided' modes but is NOT found in 'ref_only' mode?
 Pay attention to how the predicted transcripts line up with known transcripts.  Try loading the Ensembl transcripts track (File -> Load from Server)
	
Load a BAM file at the same time as the junctions.bed and merged.gtf files:
	http://cbw##.ssh01.com/rnaseq/alignments/tophat/Tumor_ALL/accepted_hits.bam
	
NOTE: We have just scratched the surface exploring these output files!!!
	
	
 ##SAVING A COPY OF YOUR RESULTS TO TAKE HOME WITH YOU
 If you are performing this tutorial on a cloud instance, everything will be deleted when the instance is destroyed!
 To package and download everything used or created during the tutorials you can do the following from your cloud terminal session.
 First package and compress all of the directories and files in the ‘rnaseq’ directory
	cd /home/ubuntu/workspace/
	tar -czvf rnaseq_tutorial.tar.gz rnaseq/
	
 Now you can download this to your own computer from here:
 http://cbw##.ssh01.com/rnaseq_tutorial.tar.gz
	
 To unpack this archive at a terminal session on your own Linux or Mac computer you can do the following:
	tar -xzvf rnaseq_tutorial.tar.gz
	
